test/net/sourceforge/kolmafia/DataFileConsistencyTest.java

```diff
@@ -120,4 +123,74 @@ public void testFamiliarSpecificEquipment() throws IOException {
           greaterThan(0));
     }
   }
+
+  @Test
+  public void testValidZonesForCombats() {
+    String file = "combats.txt";
+    int version = 1;
+    String[] fields;
+    try (BufferedReader reader = FileUtilities.getVersionedReader(file, version)) {
+      while ((fields = FileUtilities.readData(reader)) != null) {
+        if (fields.length == 1) {
+          continue;
+        }
+        // First field must be a string and a valid zone name
+        String zone = fields[0];
+        assertTrue(AdventureDatabase.validateAdventureArea(zone), "Problem with " + zone);
+      }
+    } catch (IOException e) {
+      fail("Couldn't read from combats.txt");
+    }
+  }
+
+  @Test
+  public void testValidFrequencyForCombats() {
+    String file = "combats.txt";
+    int version = 1;
+    String[] fields;
+    try (BufferedReader reader = FileUtilities.getVersionedReader(file, version)) {
+      while ((fields = FileUtilities.readData(reader)) != null) {
+        if (fields.length == 1) {
+          continue;
+        }
+        // Second field is a Combat frequency - -1 or 0 <= x <= 100
+        String freq = fields[1];
+        int iFreq = StringUtilities.parseInt(freq);
+        boolean allowed = (iFreq == -1) || ((0 <= iFreq) && (iFreq <= 100));
```

Comment:
I would just assertThat(freq, greaterThanOrEqualTo(-1)) and assertThat(freq, lessThanOrEqualTo(100))

Reply:
OK